tools/PowerShell/Microsoft.WinGet.Client/src/Library/Install-WinGetPackage.ps1

```diff
@@ -0,0 +1,174 @@
+
+        .PARAMETER Source
+        Name of the Windows Package Manager private source. Can be identified by running: "Get-WinGetSource" and using the source Name
+
+        .PARAMETER Interactive
+        Used to specify the installer should be run in interactive mode.
+
+        .PARAMETER Silent
+        Used to specify the installer should be run in silent mode with no user input.
+
+        .PARAMETER Locale
+        Used to specify the locale for localized package installer.
+
+        .PARAMETER Log
+        Used to specify the location for the log location if it is supported by the package installer.
+
+        .PARAMETER Header
+        Used to specify the value to pass as the "Windows-Package-Manager" HTTP header for a REST source.
+
+        .PARAMETER Version
+        Used to specify the Version of the package
+
+        .PARAMETER Channel
+        Used to specify the Channel for the package. Note the Windows Package Manager as of version 1.1.0 does not support channels.
+
+        .PARAMETER VerboseLog
+        Used to provide verbose logging for the Windows Package Manager.
+        
+        .PARAMETER AcceptPackageAgreement
+        Used to accept any package agreement required for the package.
+        
+        .PARAMETER AcceptSourceAgreement
+        Used to explicitly accept any agreement required by the source.
+
+        .EXAMPLE
+        Install-WinGetPackage -id "Publisher.Package"
+
+        This example expects only a single package containing "Publisher.Package" as a valid identifier.
+
+        .EXAMPLE
+        Install-WinGetPackage -id "Publisher.Package" -source "Private"
+
+        This example expects the source named "Private" contains a package with "Publisher.Package" as a valid identifier.
+
+        .EXAMPLE
+        Install-WinGetPackage -Name "Package"
+
+        This example expects the source named "Private" contains a package with "Package" as a valid name.
+    #>
+
+    PARAM(
+        [Parameter(Position=0)] $Filter,
+        [Parameter()]           $Name,
+        [Parameter()]           $Id,
+        [Parameter()]           $Moniker,
+        [Parameter()]           $Source,
+        [Parameter()] [ValidateSet("User", "Machine")] $Scope,
+        [Parameter()] [switch]  $Interactive,
+        [Parameter()] [switch]  $Silent,
+        [Parameter()] [string]  $Version,
+        [Parameter()] [switch]  $Exact,
+        [Parameter()] [switch]  $Override,
+        [Parameter()] [string]  $Location,
+        [Parameter()] [switch]  $Force,
+        [Parameter()] [ValidatePattern("^([a-zA-Z]{2,3}|[iI]-[a-zA-Z]+|[xX]-[a-zA-Z]{1,8})(-[a-zA-Z]{1,8})*$")] [string] $Locale,
+        [Parameter()] [string]  $Log, ## This is a path of where to create a log.
+        [Parameter()] [switch]  $AcceptSourceAgreements
+    )
+    BEGIN
+    {
+        [string[]] $WinGetArgs  = "Install"
+        IF($Filter){
+            $WinGetArgs += "--Manifest", $Filter
+        }
+        IF($Name){
```

Comment:
```suggestion
        IF($PSBoundParameters.ContainsKey('Name')){
```
I've run into issues with this in the past, where it may not be detected properly. You may want to consider checking if the parameter is bound rather than checking if it has a non-null value

Reply:
Same with this one @denelon 

Would apply to all the instances where you’re checking If a parameter is set